test/system/test_tpm2_akparse.sh

```diff
@@ -4,52 +4,55 @@
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
 # and/or other materials provided with the distribution.
 #
 # 3. Neither the name of Intel Corporation nor the names of its contributors
 # may be used to endorse or promote products derived from this software without
 # specific prior written permission.
 #
-# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" 
-# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
-# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
-# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE 
-# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
-# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF 
-# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
-# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
-# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
-# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 # THE POSSIBILITY OF SUCH DAMAGE.
 #;**********************************************************************;
- 
-new_path=`dirname $0`
-PATH="$PATH":"$new_path"
 
-file_input_data=ak_pub.out
-output_akparse=akparse.out
+onerror() {
+    echo "$BASH_COMMAND on line ${BASH_LINENO[0]} failed: $?"
+    exit 1
+}
+trap onerror ERR
+
+cleanup() {
+    rm -f ak.out akparse.out ek.pub ak.pub ak.name &>/dev/null
 
-rm $output_ekparse -rf 
+    # Evict persistent handles, we want them to always succeed and never trip
+    # the onerror trap.
+	tpm2_evictcontrol -Q -A o -H 0x8101000b 2>/dev/null || true
+	tpm2_evictcontrol -Q -A o -H 0x8101000c 2>/dev/null || true
+}
+trap cleanup EXIT
 
-if [ ! -e "$file_input_data" ]   
-  then    
-  test_tpm2_getpubak.sh
-fi 
+cleanup
 
- tpm2_akparse -f $file_input_data  -k $output_akparse
-if [ $? != 0 ];then
-	echo "akparse fail, please check the environment or parameters!"
-	exit 1
-fi
+tpm2_getpubek -Q -H 0x8101000b -g rsa -f ek.pub
 
-echo "akparse successfully!"
```

Comment:
I always find these messages to be useful. Yes, you could check $? but having instant feedback when running a test is a good thing IMHO.

Reply:
Having said that, I don't think "akparse successfully!" is even a correct english phrase so probably needs to be fixed. But making all the tests to output a consistent succeed message would be nice.

Reply:
test_all.sh handles that

Reply:
Yes, but no when run it as an standalone test.

Reply:
True, but I have some fancy output changes to test_all.sh...
I'd like the tests to be executable stand-alone scripts, but one can check the return for 0 for success.

Run these locally with test_all.sh and let me know what you think of the output.

Reply:
@williamcroberts cool, I didn't have time to finish reviewing all the patches in the PR today so I didn't notice those changes but the spinning thing is nice indeed.

But I don't think those are mutually exclusive, after all the wrapper doesn't care about the test output, only its exit code.

Or the wrapper could pass an argument to the tests so only print their exit status when executed as standalone. I also mentioned that one could check $? but that's not user friendly in my opinion.

Reply:
We can do that, that's simple enough